libs/features/mas/src/mas-element.js

```diff
@@ -153,6 +153,8 @@ export class MasElement {
         this.state = STATE_FAILED;
         this.update();
         this.log?.error('Failed:', { element: this.wrapperElement, error });
+        // Allow calling code to perform sync updates of this element
+        // before notifying observers about state change
```

Comment:
actually, `setImmediate` is deprecated, should we try it with `setTimeout(callback, 0)` by this occasion?